src/connection.js

```diff
@@ -753,12 +755,15 @@ class Connection extends EventEmitter {
     });
 
     if (preloginPayload.encryptionString === 'ON' || preloginPayload.encryptionString === 'REQ') {
-      if (!this.config.options.encrypt) {
+      if (this.config.options.encrypt === false) {
         this.emit('connect', ConnectionError("Server requires encryption, set 'encrypt' config option to true.", 'EENCRYPT'));
         return this.close();
+      } else if (typeof this.config.options.encrypt === 'undefined') {
```

Comment:
What do you think about changing the conditions here to be a bit more explicit?

```js
if (this.config.options.encrypt === true) {
  ...
} else if (this.config.options.encrypt === false) {
  ...
} else {
  ...
}
```

Reply:
I don't mind tbh, from what I recall (this was a while ago!) Semantically I guess it would still behave the same?